src/modules/complianceengine/src/lib/Pattern.h

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef COMPLIANCEENGINE_PATTERN_H
+#define COMPLIANCEENGINE_PATTERN_H
+
+#include <Regex.h>
+#include <Result.h>
+
+namespace ComplianceEngine
+{
+// Binds together a pattern and associated regular expression
+class Pattern
+{
+    // The pattern string
+    std::string mPattern;
+
+    // The compiled regular expression
+    regex mRegex;
+
+    // Constructs the pattern, may frow on incorrect patterns.
+    // Use Make(pattern) to get a Result<Pattern> object instead.
+    Pattern(const std::string& pattern) noexcept(false);
+
+public:
+    Pattern() = default;
+    Pattern(const Pattern&) = default;
+    Pattern(Pattern&&) = default;
+    Pattern& operator=(const Pattern& other)
+    {
+        if (&other == this)
+            return *this;
```

Comment:
nit (coding style): wrap one line block in curly braces

{
    //line
}

Here and elsewhere.

And super minor, don't worry to skip.